app/forms/flood_risk_engine/steps/base_address_form.rb

```diff
@@ -0,0 +1,131 @@
+module FloodRiskEngine
+  module Steps
+    class BaseAddressForm < BaseRedirectableForm
+
+      property :uprn
+      property :address_list, virtual: true
+
+      validates :uprn, presence: { message: I18n.t("flood_risk_engine.validation_errors.uprn.blank") }
+
+      validates :uprn, numericality: {
+        only_integer: true,
+        allow_blank: true,
+        message: I18n.t("flood_risk_engine.validation_errors.uprn.format")
+      }
+
+      # Post code is set in previous page, and must be passed in so we can create the
+      # address drop down list - need this externally set as we can't access Session in the Form
+      attr_reader :post_code
+
+      def prepopulator(options = {})
+        logger.debug("In prepopulator - #{options.inspect}")
+
+        # We assume postcode validation already occured so do not validate the format here
+        @post_code = options[:post_code] if options.key? :post_code
+      end
+
+      # This is a form reader - provides view with addresses for drop down selection
+      def address_list
+        @address_list ||= find_by_postcode
+      end
+
+      # The normal update action is no good for managing addresses, need to wire up continue button to use
```

Comment:
I need to set it to my own actions, process_address and process_postcode since they need to do quite different things to update